### PR TITLE
feat: add full text search to global search

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: [#5631](https://github.com/lightdash/lightdash/issues/5631)

### Description:

- Adds full text search to `charts`, `dashboards` and `spaces`
- Uses a custom english dictionary with no stop-words

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/database/seeds/development/05_generate_fake_data.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import { faker } from '@faker-js/faker';
```
Comment (owlas) [inline packages/backend/src/database/seeds/development/05_generate_fake_data.ts:1]: do we always want this seed for devs?

File: packages/backend/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
         "@aws-sdk/lib-storage": "^3.306.0",
         "@aws-sdk/s3-request-presigner": "^3.272.0",
         "@casl/ability": "^5.4.3",
+        "@faker-js/faker": "^8.4.0",
```
Comment (owlas) [inline packages/backend/package.json:35]: this should be in dev dependencies, we don't want to bundle into prod

File: packages/backend/src/database/seeds/development/05_generate_fake_data.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/backend/src/database/seeds/development/05_generate_fake_data.ts:1]: This seed only runs if `FAKE_DATA_LENGTH >= 1`, so unless we explicitly declare it this won't add anything

File: packages/backend/src/database/migrations/20240207133121_add-english-dictionary-with-custom-stop-words.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Knex } from 'knex';
```
Comment (owlas) [inline packages/backend/src/database/migrations/20240207133121_add-english-dictionary-with-custom-stop-words.ts:1]: you could merge into a single migration, why separate?

Comment (almeidabbm) [reply packages/backend/src/database/migrations/20240207133121_add-english-dictionary-with-custom-stop-words.ts:1]: good point, this was merged over multiple PRs which resulted in separate migrations.
also, merging them into 1 single file won't impact readability too much, doing it 💪

**Review by IrakliJani — APPROVED** [review]
🚀

**Pull request merged**